platforms/azure/modules/machine/machine.tf

```diff
@@ -37,6 +37,8 @@ locals {
   offer = "${element(split("${local.separator}",local.image),1)}"
   sku = "${element(split("${local.separator}",local.image),2)}"
   version = "${element(split("${local.separator}",local.image),3)}"
+
+  public_key = ""
```

Comment:
Again this strange unused variable

Reply:
I don't remember the reason for this, but If I recall correctly, this ensured the apply. I think it was requesting a public_key.

Reply:
Long ago....  I just tried to figure out what is going on here. It is a little bit more complicate:

The public key is required to create machines on Azure. Therefore the MachineClass for Azure (used by the Machine Controller to dynamically manage worke machines) also requires this info.

There is a local varible initialized with "" so far. It should take

`"${lookup(var.vm_info,"public_key","")}"`

Then the template var should use this varaible `${local.public_key}`

To make this work the vm module must export this key in azure/modules/vms/vms.tf (output vm_info). Add the key
`public_key = "${var.key}"`

Do you want to do this to make youre correction complete, or should I prepare a commit for this.

Unfortunatley we don't use the azure part anymore. Therefore there are often just formal change. Sorry for that.